application_sdk/clients/sql.py

```diff
@@ -111,19 +132,19 @@ def get_iam_user_token(self):
         Raises:
             ValueError: If required credentials (username or database) are missing.
         """
-        extra = parse_credentials_extra(self.credentials)
-        aws_access_key_id = self.credentials["username"]
-        aws_secret_access_key = self.credentials["password"]
-        host = self.credentials["host"]
+        extra = parse_credentials_extra(self.resolved_credentials)
+        aws_access_key_id = self.resolved_credentials.get("username")
+        aws_secret_access_key = self.resolved_credentials.get("password")
+        host = self.resolved_credentials.get("host")
         user = extra.get("username")
         database = extra.get("database")
         if not user:
             raise ValueError("username is required for IAM user authentication")
         if not database:
             raise ValueError("database is required for IAM user authentication")
 
-        port = self.credentials["port"]
-        region = self.credentials.get("region", None)
+        port = self.resolved_credentials.get("port",5432)
+        region = self.resolved_credentials.get("region", None)
```

Comment:
(nitpick) let's remove None from here as well. it already existed, but redundant.